frontend/src/api/MyRestaurantApi.tsx

```diff
@@ -44,3 +44,35 @@ export const useCreateMyRestaurant = () => {
   }
   return { createRestaurant, isLoading };
 };
+
+export const useGetMyRestaurant = () => {
+  const { getAccessTokenSilently } = useAuth0();
+
+  const getMyRestaurantRequest = async (): Promise<Restaurant> => {
+    const accessToken = await getAccessTokenSilently();
+
+    const response = await fetch(`${API_BASE_URL}/api/my/restaurant`, {
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+      },
+    });
+
+    if (!response.ok) {
+      throw new Error('Failed to get restaurant');
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/33#issuecomment-2394670085>#d60155</a><span> - 10/04/2024, 09:21 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The error handling in getMyRestaurantRequest could be more informative. Currently, it throws a generic error message which doesn't provide enough context for debugging or user feedback.
</div>
<div id="fix">
<b>Fix:</b> Enhance the error handling to include more context, such as the HTTP status code and any error message from the server. This will help in debugging and providing more informative feedback to users.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -58,7 +58,13 @@ export const useGetMyRestaurant = () => {
      });
 
      if (!response.ok) {
 -      throw new Error('Failed to get restaurant');
 +      const errorBody = await response.text();
 +      throw new Error(
 +        `Failed to get restaurant: ${response.status} ${response.statusText}. 
 +        ${errorBody ? `Server message: ${errorBody}` : ''}`
 +      );
      }
 
      return response.json();

```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged